dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/MetadataDumperTest.java

```diff
@@ -28,25 +28,54 @@
 import joptsimple.OptionException;
 import org.apache.commons.io.FileUtils;
 import org.junit.After;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
 /** @author shevek */
 @RunWith(JUnit4.class)
 public class MetadataDumperTest {
+  private static final String DEFAULT_FILENAME = "dwh-migration-test-metadata.zip";
+
+  private enum TopLevelTestFile {
+    DEFAULT_FILENAME_TEST_FILE(DEFAULT_FILENAME),
+    SAMPLE_DIR("sample_dir"),
+    TEST_ZIP("test.zip"),
+    DIR_ZIP("dir.zip"),
+    TEST("test");
+
+    final String filename;
+
+    TopLevelTestFile(String filename) {
+      this.filename = filename;
+    }
+  }
 
-  // TODO(ishmum): `testOverridesZipWithGivenName` with content check
-  private File file;
   private Main dumper = new Main(new MetadataDumper());
   private final Connector connector = new TestConnector();
-  private final String defaultFileName = "dwh-migration-test-metadata.zip";
+
+  @Before
+  public void setUp() throws IOException {
+    deleteTopLevelTestFiles();
+  }
 
   @After
   public void tearDown() throws IOException {
-    if (file == null || !file.exists()) return;
```

Comment:
Out of curiosity, which cases would cause the file to be considered null or non-existing, while it actually exists?

Reply:
The `testCreatesDefaultOutputZipInProvidedDirectory` method creates the `dir` directory, but the `tearDown` method doesn't remove it, because the `file` field is associated with a different path.

The `testFailsOnUnrecognizedDialect` method creates the default filename `dwh-migration-test-metadata.zip`, but the `tearDown` method doesn't remove it, because the `file` field is not assigned at all.

Reply:
Typically `file` would be null if setup failed, in which case it's nice to have that exception reported, and not masked by a subsequent failure in tearDown(). In detail, I'm not sure that's the exact case here, but that's how I would think about this as a defensive programming style.